config/initializers/timeout.rb

```diff
@@ -0,0 +1,12 @@
+# Lengthen timeout when executing REST API calls via rubyhorn (to Matterhorn) overriding
+# https://github.com/avalonmediasystem/rubyhorn/blob/avalon-r6/lib/rubyhorn/rest_client/common.rb#L8
+# Todo: not needed in Avalon v7.1
+module Rubyhorn::RestClient
+  module Common
+
+    def connect
+      super.connect
+      @client.open_timeout =  ENV['AVALON_MATTERHORN_TIMEOUT'].to_i if ENV['AVALON_MATTERHORN_TIMEOUT']
```

Comment:
Because [`super`'s `connect`](https://github.com/avalonmediasystem/rubyhorn/blob/master/lib/rubyhorn/rest_client/common.rb#L9) calls `login`, which calls `get`, which calls `execute` on the `RestClient::Request` object before you set the timeout, I don't think this will have any impact? Basically you need this to happen between lines 9 and 10 of super's method, so instead of calling super you need to override it entirely.

Reply:
Correct, the approach will not affect the timeout on the `login`. I'll add a bit more around my thinking. I'm uncomfortable around monkey patching and love input, let me know if the below changes your thoughts.
 
My thought process: 

 a) `login` timeout shouldn't be extended as it opens a denial of service opportunity, 
 b) if `login` fails I believe the [active job is retried](https://github.com/avalonmediasystem/avalon/blob/v6.5.0/app/jobs/active_encode_job.rb#L40), and 
 c) the timeouts occur in these two ingest requests [addMediaPackage and addMediaPackageUrl](https://github.com/avalonmediasystem/rubyhorn/blob/master/lib/rubyhorn/rest_client/ingest.rb#L22-L40)

How I believe the code works. After the super's connect method, the timeout is set for subsequent post/get calls such as the addMediaPackage above. Since the subsequent ingest rest calls (such as addMediaPackage) are the ones experiencing the timeouts, these call receive the longer timeout value.

Reply:
Ahh, I see. This makes sense, then. Maybe just add a comment explaining the rationale so that it doesn't get "fixed" later?